apps/api/v2/src/modules/auth/guards/roles/roles.guard.ts

```diff
@@ -11,43 +12,57 @@ import { Team } from "@calcom/prisma/client";
 @Injectable()
 export class RolesGuard implements CanActivate {
   private readonly logger = new Logger("RolesGuard Logger");
-  constructor(private reflector: Reflector, private membershipRepository: MembershipsRepository) {}
+  constructor(
+    private reflector: Reflector,
+    private membershipRepository: MembershipsRepository,
+    private readonly redisService: RedisService
+  ) {}
 
   async canActivate(context: ExecutionContext): Promise<boolean> {
     const request = context.switchToHttp().getRequest<Request & { team: Team }>();
     const teamId = request.params.teamId as string;
     const orgId = request.params.orgId as string;
     const user = request.user as GetUserReturnType;
     const allowedRole = this.reflector.get(Roles, context.getHandler());
+    const REDIS_CACHE_KEY = `apiv2:user:${user.id ?? "none"}:org:${orgId ?? "none"}:team:${
+      teamId ?? "none"
+    }:guard:roles:${allowedRole}`;
+    const cachedAccess = Boolean((await this.redisService.redis.get(REDIS_CACHE_KEY)) ?? "false");
```

Comment:
Using Boolean to convert "false" results in cachedAccess being true:
<img width="402" alt="Screenshot 2024-07-09 at 09 52 09" src="https://github.com/calcom/cal.com/assets/42170848/e8cdbd4a-478d-49ff-bf2d-1694c1a0f30b">

Reply:
We can use JSON.parse instead of Boolean:
<img width="303" alt="Screenshot 2024-07-09 at 09 54 39" src="https://github.com/calcom/cal.com/assets/42170848/2c4a618e-dbfb-42b3-931b-bb6a09fcc806">